Refactor Nav Component and Improve Mobile Menu

Lenindelrionicaragua opened the pull request:
---

**Description**:

- I created a new file named `mobileMenu.test.js` to make the `Nav` component less cluttered and more modular.
- Worked on the styles for `Nav`, `LanguageSwitch`, `NavBar`, and `MobileMenu.test` to better align with the design.
- Also made an effort to simplify some classes by reducing their usage, resulting in cleaner code.
- The component is now fully responsive and ready for mobile and small screen sizes.

![Captura de Pantalla 2025-01-15 a la(s) 17 58 32](https://github.com/user-attachments/assets/8420131a-dc8b-4ba1-ac2f-92a00140d32c)
![Captura de Pantalla 2025-01-15 a la(s) 17 58 24](https://github.com/user-attachments/assets/1f4d0292-cb36-43f5-8cd8-515b00380514)
![Captura de Pantalla 2025-01-15 a la(s) 17 58 10](https://github.com/user-attachments/assets/e4a9b1ff-8792-40b5-8aa9-05f72871e76b)

---

anaCode26 commented on src/__tests__/componentsTest/MobileMenu.test.js:
```diff
@@ -0,0 +1,145 @@
+      screen.getByRole("button", { name: "Close navigation menu" }),
+    ).toBeInTheDocument();
+
+    // Check if social media icons are rendered correctly
```
There is no need for so many comments, the code is clean and understandable.

Lenindelrionicaragua replied:
Okay I will remove them!  :)

anaCode26 commented on src/__tests__/componentsTest/MobileMenu.test.js:
```diff
@@ -0,0 +1,145 @@
+  it("renders correctly when menu is open", () => {
+    render(
+      <LanguageProvider>
+        <MobileMenu
```
A better name for this component would be "sidebar". But it's also okay like that

Lenindelrionicaragua replied:
I think you're right, I'll try changing the file name!

anaCode26 reviewed (changes requested):
It's a great refactor. Great Job! :)

YuejiaoShi commented on src/components/MobileMenu/MobileMenu.js:
```diff
@@ -0,0 +1,135 @@
+      }`}
+      data-testid="mobile-menu"
+      role="dialog"
+      aria-hidden={!isMenuOpen}
```
When I click the four page tags (Home to Contact) in the mobile screen, it gave me an error. Can you plz check if it's mattered or not?
![image](https://github.com/user-attachments/assets/f545f5ff-e3a6-4be3-941a-30afc1d3d4fc)

Lenindelrionicaragua replied:
I will review and fix this notice, thanks for finding it! :)

Lenindelrionicaragua replied:
This error does not appear on my console, but it seems to have appeared before, so it may be intermittent.

Lenindelrionicaragua replied:
Apparently only in deploy version... Im working on that!

YuejiaoShi reviewed (commented):
So nice improvement! Just found one tiny error in the console, otherwise, good to merge!

ZuzanC commented on src/components/Navbar/Navbar.js:
```diff
@@ -29,121 +30,60 @@ const Navbar = () => {
-          }`}
+    <nav
+      className="flex flex-col-1 w-full h-[7.18rem] sm:h-[14.37rem] justify-between items-center px-8 py-6 gap-2 shadow-md z-50"
+      aria-label="Main Navigation"
```
According to Davide and the design, there should be no shadow below navigation

Lenindelrionicaragua replied:
You're right, but Ana, who worked on the design, had added that, and I didn't know whether to remove it or not. I'll ask her. :)

Lenindelrionicaragua replied:
Thanks for noticing this!

Lenindelrionicaragua commented:
---

**Changes:**
- **Sidebar Component**: Renamed from `MobileMenu` to `SideBar`, and updated accessibility attributes (e.g., `inert`) to improve screen reader functionality. Removed the bottom shadow of the navbar.
- **Tests**: Refactored to use `SideBar` and updated test cases for the new component name and accessibility features.

---
